src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -858,6 +872,47 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
                     datafileSolrInputDocument.addField(SearchFields.IDENTIFIER, fileEntityId);
                     datafileSolrInputDocument.addField(SearchFields.PERSISTENT_URL, dataset.getPersistentURL());
                     datafileSolrInputDocument.addField(SearchFields.TYPE, "files");
+                    
+                    /* Full-text indexing using Apache Tika */
+                    if (settingsService.isTrueForKey(SettingsServiceBean.Key.SolrFullTextIndexing, false)) {
+                        StorageIO<DataFile> accessObject = null;
+                        InputStream instream = null;
+                        ContentHandler textHandler = null;
+                        try {
+                            AutoDetectParser autoParser = new AutoDetectParser();
+
+                            textHandler = new BodyContentHandler(-1);
+                            Metadata metadata = new Metadata();
+                            ParseContext context = new ParseContext();
+                            accessObject = DataAccess.getStorageIO(fileMetadata.getDataFile(), new DataAccessRequest());
+
+                            if (accessObject != null) {
+                                accessObject.open();
+                                instream = accessObject.getInputStream();
+
+                                // Try parsing the file. Note we haven't checked at all to
+                                // see whether this file is a good candidate.
```

Comment:
Do you think it would be worth checking file type and/or file size here?  I'm wondering about the possibility of a large binary file (aka - a big archive / hdf / etc file) adding lots of IO and CPU load, when they won't have any indexable data.

Reply:
Added optional size limit